### ISSUE TITLE
tcp vnc address gives errors

### ISSUE THREAD

**Issue by caduceus4** [tinyzimmer/kvdi#15]
Trying some experiment with setting socketAddr to tcp://hostname:5900 or tcp://hostname:5900/ so to have kvdi-proxy try to connect to a VNC port somewhere. When the kvdi-proxy container is pulled, it can't start its pod, saying that the mountPath is invalid. Doing a kubectl get pod pod/podname -o yaml shows that tcp://host:5900 is associated with a mountPath/volume, and as such is invalid for a mount or volume. This makes sense if we are using a unix domain socket, but not a TCP address.

**Comment by tinyzimmer** [comment]
You are right and looking back at the code [here](https://github.com/tinyzimmer/kvdi/blob/main/pkg/apis/kvdi/v1alpha1/desktoptemplate_util.go#L259) and [here](https://github.com/tinyzimmer/kvdi/blob/main/pkg/apis/kvdi/v1alpha1/desktoptemplate_util.go#L283) that constructs that sidecar I am laughing trying to figure out what I was thinking at the time.

You are rightfully trying to do something I wanted to be possible, and I think just forgot about it sometime along the way. Or when I was implementing those function I just wasn't thinking and didn't realize that a TCP address is obviously not a valid value to put in a container mount 😛.  Probably won't be a super hard fix. But you may find other strange behavior after we get past this speed bump.

**Comment by caduceus4** [comment]
ok, thx. I am sure there are other issues.. .there always are in things like this ;-)

**Comment by tinyzimmer** [comment]
I've started fixing this and will likely have images to test soon (will update here). One thing I'm realizing is that there are changes needing to be made to the base desktop images provided in this repo. I kinda assume you are building custom desktop images given the context (if using them at all), and you will likely need to rebuild them if they are based on these.

For details if you are curious:

 - The `/usr/local/sbin/init` script was assuming a UNIX socket also. Not in a breaking way, but in a way that will create garbage directories if it's actually a TCP socket. Putting down a somewhat hacky fix for now but it'll do.
 - At some point in time I decided that whatever you give for `socketAddr` is what you want `XVnc` to try to listen on. Can't really recall my thought process at the time, but this seems counter-intuitive to what you are trying to accomplish. To fix the above issue, the hack right now is basically just to use a generic UNIX address in the run dir if the one configured is TCP. I think at the very least, you might want to disable the `display.service` in your desktop image, if relevant for your use case. But it may not matter.

**Comment by tinyzimmer** [comment]
I'm gonna try to bundle a few things into the next tag. But for this particular issue, you can test against the latest image when [this build](https://github.com/tinyzimmer/kvdi/runs/1837278791?check_suite_focus=true) is finished. The fixes are in the `manager` deployment. If you installed with helm calling the release "kvdi", you can do something like this:

```sh
kubectl patch deployment kvdi-manager --patch \
    '{"spec":{"template":{"spec":{"containers":[{"name":"kvdi-manager","image":"ghcr.io/tinyzimmer/kvdi:manager-latest"}]}}}}'
```

**Comment by tinyzimmer** [comment]
The change is actually released now along with other stuff - you can check the notes [here](https://github.com/tinyzimmer/kvdi/releases/tag/v0.1.4)

**Comment by caduceus4** [comment]
ok thanks.. i've been off doing other things and now circling back. yes we are building our own desktops...

**Comment by tinyzimmer** [comment]
Haha no worries. I've been back and forth and I've actually made some pretty big changes since this first got posted. Hopefully the release notes and other threads will be informative enough, but let me know if you have trouble. 

The next thing I want to release could also affect you. I'm sorta completely rewriting the proxy to be a simple TCP server instead of a full blown web server. Some of the environment variables and stuff used by the inits are changing too.

**Comment by caduceus4** [comment]
so far so good using the tcp address for sockAddr.. but not yet finished testing
